File: jphp-game-ext/src/main/resources/JPHP-INF/sdk/php/game/UXGameScene.php
<?php
namespace php\game;
use php\gui\layout\UXAnchorPane;

/**
 * Class UXGameScene
 * @package php\game
 */
class UXGameScene
{
    /**
     * @readonly
     * @var UXAnchorPane
     */
    public $content;

    /**
     * @var bool
     */
    public $physicsEnabled = true;

    /**
     * UXGameScene constructor.
     * @param UXAnchorPane $content
     */
    public function __construct(UXAnchorPane $content)
    {
    }

    /**
     * @param UXGameEntity $entity
     */
    public function add(UXGameEntity $entity)
    {
    }

    /**
     * @param UXGameEntity $entity
     */
    public function remove(UXGameEntity $entity)
    {
    }
}

File: jphp-game-ext/src/main/java/org/develnext/jphp/ext/game/classes/UXGameScene.java
package org.develnext.jphp.ext.game.classes;

import org.develnext.jphp.ext.game.GameExtension;
import org.develnext.jphp.ext.game.support.GameEntity2D;
import org.develnext.jphp.ext.game.support.GameScene2D;
import org.develnext.jphp.ext.game.support.Vec2d;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.lang.BaseWrapper;
import php.runtime.reflection.ClassEntity;

@Reflection.Namespace(GameExtension.NS)
public class UXGameScene extends BaseWrapper<GameScene2D> {
    interface WrappedInterface {
        @Property Vec2d gravity();

        void play();
        void pause();
    }

    public UXGameScene(Environment env, GameScene2D wrappedObject) {
        super(env, wrappedObject);
    }

    public UXGameScene(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct() {
        __wrappedObject = new GameScene2D();
    }

    @Signature
    public void add(GameEntity2D entity) {
        getWrappedObject().addEntity(entity);
    }

    @Signature
    public void remove(GameEntity2D entity) {
        getWrappedObject().removeEntity(entity);
    }
}
